content/ui-patterns/saving.mdx

```diff
@@ -0,0 +1,157 @@
+*/}
+
+### Declarative inputs
+
+When designing a form that is saved explicitly, use the following inputs:
+- Text inputs
+- Checkbox groups
+- Radio button groups
+- Dropdown menus
+
+To maintain consistent interaction patterns across your app, do not use toggle switches or segmented controls in an explicitly saved form. They should only be used when they automatically change based on user input.
+
+## Automatic saving
+
+Automatic saving applies changes as they are made.
+
+Automatic saving should be used when the user expects instant feedback from the change they made. Changes should only be saved automatically using[imperitive inputs](#imperitive-inputs) and for individual settings that are not changed via separate flows.
+
+If a field is saved automatically, it should be obvious whether or not it saved without using text or visual indicators. A visual indicator for a field that has been "saved" or has an error "error" could be mistaken as a validation status.
+
+![Screen recording of a user changing their theme in the 'Appearance' settings tab](https://user-images.githubusercontent.com/2313998/150868921-afc2b8b8-e39f-478f-b17d-398b790bda89.gif)
+
+
+### Imperitive inputs
+
+When user input is automatically saved, use the following inputs:
+- Toggle switches
+- Segmented controls
+- Single-select dropdowns
+
+These inputs are designed to reduce UI by giving instant feedback. The toggle switch behaves like a light switch: the light just switches on and off without you having to confirm each time you flip the switch. Segmented controls and single-select dropdowns behave like a soda machine: the soda comes out as soon as you press the buttons without you confirming which soda you wanted.
+
+To build an accessible and consistent app, do not automatically save declarative inputs.
+
+### Inputs that are problematic for autosaving:
+**Text inputs**
+- Users are expecting to set a value and then submit
+- Sensitive information could be unintentionally submitted. For example: a change password input
+
+**Checkbox groups**
+- It could be unclear whether or not their selections have been saved
+- Users may accidentally submit a selection that they clicked by accident
+
+**Radio button groups**
+Screenreader users can't read the options without selecting them. This could accidentally apply a selection the user didn't want.
+
+**Browser-native `<select>` dropdowns**
+On Windows, a select dropdown has a similar problem as a radio button group. 
+Options are selected when a user focuses the select and uses their arrow keys to read through each option
+
+## Feedback
+
+If the change may not be obvious in the UI within the user's viewport, provide feedback to let the user know that the change occurred successfully. If the change occured without a redirect or page refresh, you could use a toast message. If the change occured and the page refreshes or redirects, you could use a flash banner message.
```

Comment:
We might have to not recommend toast messages until we fix the accessibility issues.

Reply:
Are the accessibility issues caused by the frontend implementation? Or are we designing with toasts in a way that is not accessible?

If it's just about the frontend implementation, we could keep the guidance but add a note that toasts should be avoided until we fix them.